tests/lib/openstack_query/runners/test_runner_wrapper.py

```diff
@@ -75,146 +75,260 @@ def _run_paginated_query_test(number_iterations):
 # pylint:disable=too-many-arguments
 
 
-@patch("openstack_query.runners.runner_wrapper.RunnerWrapper._apply_client_side_filter")
-@patch("openstack_query.runners.runner_wrapper.RunnerWrapper._run_query")
-@patch("openstack_query.runners.runner_wrapper.RunnerWrapper._parse_meta_params")
-def test_run_with_only_client_side_filter(
-    mock_parse_meta_params,
-    mock_run_query,
-    mock_apply_client_side_filter,
-    instance,
-    mock_connection,
-    mock_openstack_connection,
-):
+@pytest.fixture(name="runner_run_test_case")
+def runner_run_test_case_fixture(instance, mock_connection, mock_openstack_connection):
     """
-    Tests that run method functions expectedly - with only client_side_filter_func set
-    method should call run_query and run apply_client_side_filter and return results
-    also tests with no meta-params set
+    Fixture to run run() test cases with different args
     """
 
-    mock_run_query.return_value = ["openstack-resource-1", "openstack-resource-2"]
-    mock_apply_client_side_filter.return_value = ["openstack-resource-1"]
-
-    mock_client_side_filter_func = MagicMock()
-    mock_cloud_domain = MagicMock()
+    def _runner_run_test_case(
+        mock_client_side_filters=None,
+        mock_server_side_filters=None,
+        mock_from_subset=None,
+        **mock_kwargs,
+    ):
+        """
+        method to run run() test case with provided input values
+        The individual methods that are called in run must be patched out by the test function
+        prior to this and any asserts need to be done by the test function
+        """
 
-    res = instance.run(
-        cloud_account=mock_cloud_domain,
-        client_side_filter_func=mock_client_side_filter_func,
+        with patch(
```

Comment:
I want to come in and refactor run in the future, so we'll get this in as-is and I'll throw together a prototype refactor :thinking:

Reply:
please do